### ISSUE TITLE
Set-cookie header is not sent in the response

### ISSUE THREAD

**Issue by keerthivasan-r** [hapijs/hapi#3197]
Hi,
I have configured the state options in the hapi server using

```
   // configure device cookie
    server.state('device', {
        ttl: 24 * 60 * 60 * 1000 * 365,   // 1 year
        isSecure: false,
        path: '/',
        encoding: 'base64json'
        // TODO: set password and inspect API
    });
```

I'm setting the cookie by 

```
  return reply({message: 'device registered, device-cookie was set'})
                    .state('device', {device_id: result.deviceId}).code(201);
```

There is no error at all. I have also enabled cors hapi-cors-header plugin. But, i couldn't see the cookie under resources in chrome inspect tool. I get the response message and response code - 201. but, the cookie is not sent in the response. Am i missing something? This is a cross domain POST request

The client runs on 5555 and the hapi server running on 8200. I tried to debug, but couldn't get much information. Can someone shed some light on it

**Comment by mtharrison** [comment]
Are you setting with XHR `withCredentials` flag on the client and did you set the `credentials` option under the route `cors` option?

**Comment by keerthivasan-r** [comment]
Yes, you are right. but, i haven't set the credentials option previously under routes. Now, i have made it this way.

```
if (!process.env.HOST) {
    manifest["connections"] = [{
        host: 'localhost',
        port: 8200,
        routes: {
            cors: {
                origin : ['*'],
                credentials : true
                }
        }

    }];
}

Still, no luck. it didn't work!

```

**Comment by mtharrison** [comment]
That looks a bit strange. Afaik `Access-Control-Allow-Credentials` is a response header only, why is it in the list of request headers?

Too tricky to guess what's happening here. Can you make and post an [MCVE](http://stackoverflow.com/help/mcve)? Or at least show the code that makes the request?

**Comment by keerthivasan-r** [comment]
I'm sorry that was my mistake. I added it in the request. I have removed it from the request sending code.

I also fixed the problem, Hapi was perfectly returning the cookies in the response. I just missed to set the credentials property while sending the request.

http://stackoverflow.com/questions/35007572/angular2-xhrfields-withcredentials-true/

Thanks for the support and help. Happy coding! Hapi community :)
